modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -104,7 +123,7 @@ def match_zone(fqdn)
           weight = zone_weight
         end
       end
-      raise Proxy::Dns::NotFound.new("The DNS server has no authoritative zone for #{fqdn}") unless matched_zone
+      raise Proxy::Dns::NotFound.new("The DNS server has no authoritative zone for #{fqdn_or_ip}") unless matched_zone
```

Comment:
Using IP's with mach_zone should also solve the the problem of IP not in subnet since we would not have mached_zone here, @witlessbird